packages/core/CalendarManager.ts

```diff
@@ -48,16 +56,29 @@ export const getConnectedCalendars = async (
         }
         const cals = await calendar.listCalendars();
         const calendars = _(cals)
-          .map((cal) => ({
-            ...cal,
-            readOnly: cal.readOnly || false,
-            primary: cal.primary || null,
-            isSelected: selectedCalendars.some((selected) => selected.externalId === cal.externalId),
-            credentialId,
-          }))
+          .map((cal) => {
+            if (cal.primary) {
+              primary = { ...cal, credentialId };
+            }
+            if (cal.externalId === destinationCalendarExternalId) {
+              destinationCalendar = cal;
+            }
+            return {
+              ...cal,
+              readOnly: cal.readOnly || false,
+              primary: cal.primary || null,
+              isSelected: selectedCalendars.some((selected) => selected.externalId === cal.externalId),
+              credentialId,
+            };
+          })
           .sortBy(["primary"])
           .value();
-        const primary = calendars.find((item) => item.primary) ?? calendars.find((cal) => cal !== undefined);
```

Comment:
Thinking about efficiency, since we're already looping through all calendars once, might as well see if they are the primary as well